dataforest/processes/core/TreeProcessMethods.py

```diff
@@ -53,26 +61,50 @@ def distributed_method(
                     after process execution to save memory. If boolean, whether
                     or not to clear all data attrs, if list, names of data
                     attrs to clear
+                force_rerun: force the process to rerun even if already done
+                parallel: whether or not to parallelize
                 **kwargs:
 
             Returns:
 
             """
+            kwargs = {"stop_on_error": stop_on_error, "stop_on_hook_error": stop_on_hook_error, **kwargs}
             if not self._branch_cache.fully_loaded:
                 self._branch_cache.load_all()
-            return_vals = []
-            for branch in list(self._branch_cache.values()):
+            all_branches = list(self._branch_cache.values())
+            unique_branches = {str(branch.spec[:method_name]): branch for branch in all_branches}
+
+            def _single_kernel(branch):
                 branch_method = getattr(branch.process, method_name)
-                if not branch[method_name].done:
-                    return_vals.append(
-                        branch_method(
-                            *args, stop_on_error=stop_on_error, stop_on_hook_error=stop_on_hook_error, **kwargs
-                        )
-                    )
+                if not branch[method_name].done or force_rerun:
+                    ret = branch_method(*args, **kwargs)
                     if clear_data:
                         clear_kwargs = {"all_data": True} if isinstance(clear_data, bool) else {"attrs": clear_data}
                         branch.clear_data(**clear_kwargs)
-            return return_vals
+                    return ret
+
+            def _distributed_kernel_serial():
+                _ret_vals = []
+                for branch in unique_branches.values():
+                    _ret_vals.append(_single_kernel(branch))
+                return _ret_vals
+
+            def _distributed_kernel_parallel():
+                process = delayed(_single_kernel)
+                pool = Parallel(n_jobs=self._N_JOBS)
```

Comment:
If we have 2 cores, wouldn't this use only one core since `_N_JOBS = cpu_count() - 1`? https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html